src/main/java/io/github/moulberry/notenoughupdates/profileviewer/BingoPage.java

```diff
@@ -0,0 +1,311 @@
+				"Top " + EnumChatFormatting.YELLOW + "25%" + EnumChatFormatting.DARK_GRAY + " - " + EnumChatFormatting.GOLD +
+					"7 Bingo Points");
+			tooltip.add(
+				"All Contributors" + EnumChatFormatting.DARK_GRAY + " - " + EnumChatFormatting.GOLD + "4 Bingo Points");
+
+			tooltip.add("");
+			tooltip.add(EnumChatFormatting.DARK_GRAY.toString() + EnumChatFormatting.ITALIC + "Community Goals are");
+			tooltip.add(
+				EnumChatFormatting.DARK_GRAY.toString() + EnumChatFormatting.ITALIC + "collaborative - anyone with a");
+			tooltip.add(
+				EnumChatFormatting.DARK_GRAY.toString() + EnumChatFormatting.ITALIC + "Bingo profile can help to reach");
+			tooltip.add(EnumChatFormatting.DARK_GRAY.toString() + EnumChatFormatting.ITALIC + "the goal!");
+			tooltip.add("");
+			tooltip.add(EnumChatFormatting.DARK_GRAY.toString() + EnumChatFormatting.ITALIC + "The more you contribute");
+			tooltip.add(EnumChatFormatting.DARK_GRAY.toString() + EnumChatFormatting.ITALIC +
+				"towards the goal, the more you");
+			tooltip.add(EnumChatFormatting.DARK_GRAY.toString() + EnumChatFormatting.ITALIC + "will be rewarded");
+
+			if (finalTier == totalTiers) {
+				tooltip.add("");
+				tooltip.add(EnumChatFormatting.GREEN + "GOAL REACHED");
+			}
+		} else {
+			tooltip.add(EnumChatFormatting.GREEN + goal.get("name").getAsString());
+			tooltip.add(EnumChatFormatting.DARK_GRAY + "Personal Goal");
+			tooltip.add("");
+			tooltip.add(goal.get("lore").getAsString());
+			tooltip.add("");
+			tooltip.add(EnumChatFormatting.GRAY + "Reward");
+			tooltip.add(EnumChatFormatting.GOLD + "1 Bingo Point");
+			if (completed) {
+				tooltip.add("");
+				tooltip.add(EnumChatFormatting.GREEN + "GOAL REACHED");
+			} else {
+				tooltip.add("");
+				tooltip.add(EnumChatFormatting.RED + "You have not reached this goal!");
+			}
+		}
+		return tooltip;
+	}
+
+	private static void showMissingDataMessage(int guiLeft, int guiTop) {
+		String message = EnumChatFormatting.RED + "No Bingo data for current event!";
+		Utils.drawStringCentered(message, Minecraft.getMinecraft().fontRendererObj,
+			guiLeft + 431 / 2f, guiTop + 101, true, 0
+		);
+	}
+
+	private static List<String> jsonArrayToStringList(JsonArray completedGoals) {
+		List<String> list = new ArrayList<>();
+		for (JsonElement completedGoal : completedGoals) {
+			list.add(completedGoal.getAsString());
+		}
+		return list;
+	}
+
+	private static List<JsonObject> jsonArrayToJsonObjectList(JsonArray goals) {
+		List<JsonObject> list = new ArrayList<>();
+		for (JsonElement goal : goals) {
+			list.add(goal.getAsJsonObject());
+		}
+
+		return list;
+	}
+
+	private static void processBingoResources() {
+		long currentTime = System.currentTimeMillis();
+
+		//renew every 2 minutes
+		if (currentTime - lastResourceRequest < 120 * 1000 && bingoGoals != null) return;
+		lastResourceRequest = currentTime;
+
+		HashMap<String, String> args = new HashMap<>();
+		NotEnoughUpdates.INSTANCE.manager.hypixelApi.getHypixelApiAsync(
```

Comment:
consider using `getHypixelApiAsync(String, String, HashMap<String, String>)` (the `CompletableFuture`), and checking for `CompletableFuture#isDone()`. this approach is also okay if it has to be this way.